File: product-catalog-service/src/main/java/com/grocery/productcatalogservice/service/ProductService.java
package com.grocery.productcatalogservice.service;

import com.grocery.productcatalogservice.dto.*;

import java.util.List;

public interface ProductService {
    CategoryResponseDto createCategory(CategoryRequestDto categoryRequestDto);
    List<CategoryResponseDto> getAllCategories();
    CategoryResponseDto getCategoryById(Long categoryId);
    CategoryResponseDto updateCategory(Long categoryId,CategoryRequestDto categoryRequestDto);
    void deleteCategory(Long categoryId);

    ProductResponseDto createProduct(ProductRequestDto productRequestDto);
    List<ProductResponseDto> getAllProducts();
    ProductResponseDto getProductById(Long productID);
    ProductResponseDto updateProduct(Long productID, ProductRequestDto productRequestDto);
    void deleteProduct(Long productID);

}

File: frontend/src/services/productService.js
import api from "./api";

export const getAllProducts = () => {
  return api.get("/api/products");
};

export const getProductsById = (productId) => {
  return api.get(`/api/products/${productId}`);
};

export const createProduct = (product) => {
  return api.post("/api/products", product);
};

export const updateProduct = (productId, product) => {
  return api.put(`/api/products/${productId}`, product);
};

export const deleteProduct = (productId) => {
  return api.delete(`/api/products/${productId}`);
};
